### PR TITLE
Extract and refactor client

### PR DESCRIPTION
**Pull request by fyurchik**
* [x] Extract client class 
* [x] Refactor client class

### CODE REVIEW

File: lib/rubyai/client.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def body
+      {
+        'model': @model,
+        'messages': [{"role": "user", "content": @messages}],
+        'temperature': @temperature
+      }
+    end
+
+    def connection
```
Comment (fyurchik) [inline lib/rubyai/client.rb:38]: do we need a memorization here? like:
``` ruby
def connection
  @connection ||= Faraday.new do |faraday|
     faraday.adapter Faraday.default_adapter
  end
end 
```

File: lib/rubyai/client.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        req.url BASE_URL
+        req.headers['Content-Type'] = 'application/json'
+        req.headers['Authorization'] = "Bearer #{@api_key}"
+        req.body = body.to_json
+      end
+
+      JSON.parse(response.body)
+    end
+
+    def mode(model)
```
Comment (alexshapalov) [inline lib/rubyai/client.rb:39]: In Ruby, when you declare attr_accessor :model, it automatically creates both a getter and a setter method for :model. This means you have the following methods implicitly defined by that single line:

```
def model  # This is the getter
  @model
end

def model=(value)  # This is the setter
  @model = value
end
```

Comment (alexshapalov) [reply lib/rubyai/client.rb:39]: you can delete this method

**Review by alexshapalov — CHANGES_REQUESTED** [review]

File: lib/rubyai/client.rb (hunk 2 above)
Comment (fyurchik) [reply lib/rubyai/client.rb]: done

**Pull request merged**